### PR TITLE
Algorithm 2 & general code improvements

### PR DESCRIPTION
**Pull request by wouwouwou**

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -111,9 +112,122 @@ def compute_mutual_information(training_set, event1, event2):
...
+
+    # While there are still vertexes to check
+    while len(checked_vertexes) < len(result.vertices):
+
+        # Pop one of the vertexes to check
+        v = vertexes_to_check_edge_orientation.pop()
+
+        # Check every edge incident to v and add it to the vertexes which need to be checked.
+        for e in v.incidence:
+            if not checked_vertexes.__contains__(e.other_end(v)):
```
Comment (Vescatur) [inline main.py]: Why not use e.other_end(v) not in check_vertexes?

File: main.py [hunk 2]
```diff
@@ -111,9 +112,122 @@ def compute_mutual_information(training_set, event1, event2):
...
+def check_potential_articulation_point_fault_tree(parent_edge: Edge):
+    parent = parent_edge.head
+    articulation_point = parent_edge.tail
+    for e in articulation_point.incidence:
+        child = e.other_end(articulation_point)
+        if not child == parent:
+            determine_independence_of_parent_and_child(parent, articulation_point, child)
+
+
+def get_potential_articulation_point_from_just_oriented_edge(edge: Edge, parent: Vertex):
```
Comment (Vescatur) [inline main.py]: Never called

File: main.py [hunk 3]
```diff
@@ -111,9 +112,122 @@ def compute_mutual_information(training_set, event1, event2):
...
+    """
+    articulation_vertex = edge.other_end(parent)
+    result = list()
+    for e in articulation_vertex.incidence:
+        if not e.other_end(articulation_vertex) == parent:
+            result += (edge, e)
+    return result
+
+
+def check_edge_direction_event_tree(edge: Edge, origin_vertex: Vertex):
```
Comment (Vescatur) [inline main.py]: The name check implies only a read operation. I would call something allong the lines of swap

File: main.py (hunk 2 above)
Comment (wouwouwou) [reply main.py]: Removed

File: main.py (hunk 3 above)
Comment (wouwouwou) [reply main.py]: Changed the method name

File: main.py (hunk 1 above)
Comment (wouwouwou) [reply main.py]: Good one, changed

**Review by MartijnVerkleij — APPROVED** [review]

**Pull request merged**